syntax = "proto3";
package CentralAuthenticationService;

service AccountService {
  rpc LogOut (Empty) returns (Empty);
  rpc GetProfile (Account) returns (Account);
  rpc UpdateProfile (Account) returns (Account);
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc SignUp(SignUpRequest) returns (SignUpResponse);
}

message Account {
  string id = 1;
  string username = 2;
  string phone = 4;
  string email = 5;
  string password = 6;
}

message Token {
  string access_token = 1;
  string refresh_token = 2;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string token = 1;
}

message SignUpRequest {
    string username = 1;
    string email = 2;
    string password = 3;
}

message SignUpResponse {
    string token = 1;
}


message Empty { }
